File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    answer: int = 1
    routes.sort(key=lambda x: x[1])
    camera_point: int = routes[0][1]
    
    for i in range(1, len(routes)):
        start: int
        end: int
        start, end = routes[i]
        
        if camera_point < start:
            answer += 1
            camera_point = end
    
    return answer

File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        Arrays.sort(routes, (a, b) -> Integer.compare(a[1], b[1]));
        int answer = 1;
        int camera = routes[0][1];
        for (int i = 1; i < routes.length; i++) {
            if (routes[i][0] > camera) {
                answer++;
                camera = routes[i][1];
            }
        }
        return answer;
    }
}